enhancements/olm/olm-managed-operator-metrics.md

```diff
@@ -0,0 +1,130 @@
+
+A number of requirements must be met for an applications metrics to be scraped by the OpenShift Monitoring Prometheus Operator.
+
+##### Namespace Requirements
+
+The application must be deployed in a namespace that:
+
+- Is prefixed with `openshift-`
+- Is labeled with `openshift.io/cluster-monitoring=true`
+
+##### ServiceMonitor Requirements
+
+A `ServiceMonitor` object must be created in the namespace mentioned above and point to the application's metrics endpoint.
+
+>Note: This requirement is already fulfilled with OLM bundle support for Prometheus resources.
+
+##### RBAC Requirements
+
+A role/rolebinding must be created that provides the Prometheus Operator ServiceAccount with the appropriate RBAC privileges to discover the `ServiceMonitor` and `PrometheusRule` resources in the newly created namespace.
+
+#### Fulfilling Namespace and RBAC Requirements
+
+OLM itself will not be responsible for fulfilling the Namespace and RBAC requirements. Instead, OLM will define how operator authors can provide OpenShift Console with the information required to fulfill these requirements.
+
+OLM [building-your-csv](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md) documentation will be updated to describe two new supported annotations:
+
+- The `operatorframwork.io/suggested-namespace` annotation. When the `operatorframwork.io/suggested-namespace` annotation is present the UI will highlight the suggested namespace when installing the operator. OLM itself will not require that the operator is deployed in the namespace defined by the `operatorframwork.io/suggested-namespace` annotation.
+
+- The `operatorframwork.io/cluster-monitoring=true` annotation. When this annotation is set to `true`, the OpenShift Console will update the namespace that the operator is being deployed to with the `openshift.io/cluster-monitoring=true` label. When this annotation is present, the UI will update the OpenShift Monitoring Prometheus Operator ServiceAccount with the appropriate RBAC privileges for the given namespace as well, allowing operators to be scraped by the OpenShift Monitoring Prometheus Operator.
+
+>Note: There is no work required by the OLM team to implement this support.
+
+#### Risks and Mitigations
+
+If OLM allows any operator to report metrics to the OpenShift Monitoring Prometheus Instance, there is a chance that the prometheus instance could be overloaded or the integrity of the data it scraps could be jeopardize. In an effort to minimize this risk, the number of operators that are granted permission to include metrics will be highly regulated. Automation will be added to the [community-operators](https://github.com/operator-framework/community-operators/)` repository to ensure that any operators that introduce metrics are manual reviewed and approved.
```

Comment:
Do we want to mention in this section, the e2e test job we discussed needs to be added and possible tests in each approved red hat operator as well, to prevent cardinality explosions.